common/base/include/claraparabricks/genomeworks/utils/mathutils.hpp

```diff
@@ -78,6 +78,37 @@ std::int32_t int_floor_log2(T val)
     return power;
 }
 
+/// @brief Rounds up a number to the next number divisible by the given denominator. If the number is already divisible by the denominator it remains the same
+/// @param val number to round up
+/// @param roundup_denominator has to be positive
+/// @tparam Integer has to be integer
+template <typename Integer>
+__host__ __device__ Integer roundup_next_multiple(const Integer val,
+                                                  int32_t roundup_denominator)
+{
+    static_assert(std::is_integral<Integer>::value, "Expected an integer");
+    assert(roundup_denominator >= 0);
```

Comment:
This must be `assert(roundup_denominator > 0);`.

Reply:
Oh, yes...
done